### DISCUSSION TITLE
Lazer can't import beatmaps in nested folders from stable if extra file is present

### DISCUSSION THREAD

**Discussion by PercyDan54** [ppy/osu#24989] [Q&A] [answered]
When beatmaps are placed in nested folders in Stable (e.g. `Songs\SomeFolder`)
If there is an extra file in `SomeFolder` the import finishes but with all beatmap difficulties in SomeFolder "combined" to one beatmapset and lazer complains about missing audio file

Reproduction:
1. Place stable beatmaps (can be any beatmap) in nested folders and create an extra file
![image.png](https://media.discordapp.net/attachments/1097318920991559880/1158324535397978172/image.png)
2. Run the import from stable wizard.
3. Observe beatmaps placed like this
(The "cresent moon" has difficulties from "cresent moon" + "Kuchibue Jietto", with missing audio and background. You can still play the beatmap. The other map, "piece of azure", is completely normal.)
![osu_2023-10-02_17-05-58](https://github.com/ppy/osu/assets/50285552/1c43bf36-eea0-418b-95b7-d2c3150cd70d)

Resulting db file:
[client.zip](https://github.com/ppy/osu/files/12781286/client.zip)

**Comment by peppy** [comment]
This is how the importer is implemented, correct. It's done this way for simplicity.

Could you better describe your specific use case? What was the "extra file"?

**Reply by bdach** [reply]
As per https://discord.com/channels/188630481301012481/1097318920991559880/1158323499153227877, it was apparently a readme sort of thing in a beatmap pack.

**Reply by peppy** [reply]
I guess we can change the importer to add all directories as processable until we come across an `.osu` file.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/pull/24998
